packages/wrangler/src/pages/deploy.tsx

```diff
@@ -73,76 +76,146 @@ export function Options(yargs: CommonYargsArgv) {
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
 	}
 
-	if (!directory) {
-		throw new FatalError("Must specify a directory.", 1);
+	let config: Config | undefined;
+	const configPath = findWranglerToml(process.cwd(), false);
+
+	try {
+		/*
+		 * this reads the config file with `env` set to `undefined`, which will
+		 * return the top-level config. This contains all the information we
+		 * need from now. We will perform a second config file read later
+		 * in `/api/pages/deploy`, that will get the environment specififc config
+		 */
+		config = readConfig(configPath, { ...args, env: undefined }, true);
+	} catch (err) {
+		if (
+			!(
+				err instanceof FatalError && err.code === EXIT_CODE_INVALID_PAGES_CONFIG
+			)
+		) {
+			throw err;
+		}
 	}
 
-	const config = getConfigCache<PagesConfigCache>(PAGES_CONFIG_CACHE_FILENAME);
-	const accountId = await requireAuth(config);
+	// const isPagesConfig = isPagesWranglerToml(configPath);
 
-	projectName ??= config.project_name;
+	/*
+	 * If we found a `wrangler.toml` config file that doesn't specify
+	 * `pages_build_output_dir`, we'll ignore the file, but inform users
+	 * that we did find one, just not valid for Pages.
+	 */
+	if (configPath && config === undefined) {
+		logger.warn(
+			`Pages now has wrangler.toml support.\n` +
+				`We detected a configuration file at ${configPath} but it is missing the "pages_build_output_dir" field, required by Pages.\n` +
+				`If you would like to use this configuration file to deploy your project, please use "pages_build_output_dir" to specify the directory of static files to upload.\n` +
+				`Ignoring configuration file for now, and proceeding with project deploy.`
+		);
+	}
 
-	const isInteractive = process.stdin.isTTY;
-	if (!projectName && isInteractive) {
-		const projects = (await listProjects({ accountId })).filter(
-			(project) => !project.source
+	const directory = args.directory ?? config?.pages_build_output_dir;
+	if (!directory) {
+		throw new FatalError(
+			"Must specify a directory of assets to deploy. Please specify the [<directory>] argument in the `pages deploy` command, or configure `pages_build_output_dir` in your `wrangler.toml` configuration file.",
+			1
 		);
+	}
+
+	const configCache = getConfigCache<PagesConfigCache>(
+		PAGES_CONFIG_CACHE_FILENAME
+	);
+	const accountId = await requireAuth(configCache);
+
+	let projectName =
+		args.projectName ?? config?.name ?? configCache.project_name;
+	let isExistingProject = true;
 
+	if (projectName) {
+		try {
+			await fetchResult<Project>(
+				`/accounts/${accountId}/pages/projects/${projectName}`
+			);
+		} catch (err) {
+			// code `8000007` corresponds to project not found
+			if ((err as { code: number }).code !== 8000007) {
+				throw err;
+			} else {
+				isExistingProject = false;
+			}
+		}
+	}
+
+	const isInteractive = process.stdin.isTTY;
+	if (
+		(!projectName || (projectName !== undefined && !isExistingProject)) &&
```

Comment:
this feels a bit convoluted... do we need to distinguish undefined from, say, empty string? Could it just be

```suggestion
		(!projectName || !isExistingProject) &&
```